### ISSUE TITLE
Philips Model 43PUS8518/12 (Android TV / Chanaellist Version 125)

### ISSUE THREAD

**Issue by dssHess** [PredatH0r/ChanSort#420]
Hello
I have a new Philips TV
Model 43PUS8518/12
SW version TPM231WW_R.101.001.227.225
This is a TV with Android.
Before the last firmware update, the channel lists had version 120. Now it is 125.
Chartsort says when loading that it cannot process version 125. Chansort works perfectly with version 120.
What can I do? Or how can you help?
Kind regards
Thomas

**Comment by PredatH0r** [comment]
Hi Thomas!

Can you please send me your current channel list? So far I have not seen any files with version 125.
Out of curiosity: Did you use ChanSort with version 120 before? Did that work for you?
Most users reported that the TV will ignore the modified (main) channel numbers and only the favorite list is imported with the new numbers - and even there are supposedly some caviats.
Unfortunately (?) I don't have a Philips TV myself to test this out.

Best regards,
Horst

**Comment by dssHess** [comment]
Hello Horst,
I have attached the channel list as a ZIP file.

H: Have you ever used ChanSort with version 120? Did it work for you?

T: I loaded the version 120 list into Chartsort without any problems. I also edited it. But I haven't got around to saving it back to the device yet and the firmware has changed in the meantime.
So I can't say anything more yet.
But if Chartsort can read the list, I'll be happy to share my results and any additional information with you.

Best regards,
Thomas
[E.zip](https://github.com/user-attachments/files/16660083/E.zip)

**Comment by Liro3000** [comment]
Hello,
I am experiencing the same issue. I have a Philips 65OLED889 with channel version 125. Is there any solution to fix this yet?

**Comment by alien2000a** [comment]
I also have this problem on 2023 Philips 8808 TV. We've had a mail correspondence on this and I've sent my channel list as requested on 29/7/2024. 

I'm attaching my file list once again in case you still need it.

Please Horst don't leave us alone with this. Thank you in advance for looking into it. All the 
[PhilipsChannelMaps.zip](https://github.com/user-attachments/files/17099519/PhilipsChannelMaps.zip)
best.

**Comment by PredatH0r** [comment]
Thanks for all your feedback. I haven't had time too look into this format in detail yet.
With the current weather forecast for the upcoming weekend there is a fair chance I can get to it.

**Comment by PredatH0r** [comment]
https://github.com/PredatH0r/ChanSort/releases/tag/v2024-10-01

Editing of the favorites list is confirmed to work.

Editing of the main channel numbers hasn't been tested yet since I don't have a Philips TV. This new version edits the DVB*.xml files and the new MtkChannelList.xml file in a synchronized way. Let's hope it works.

Any feedback is appreciated.

**Comment by bogdan-dmd** [comment]
I have the same TV as in the topic title. Tried using chansort only now after the 125 update. I'm encountering the following problems:
- I can only open MtkChannelList.xml. All the other xml's and bin file give me the following error when I try to open them: "MtkChannelList.xml doesn't contain a matching channel for @12: DIGI 4K" (DIGI 4K is the first channel on the tv's list). For all these files, except for MtkChannelList.xml, I also get the Philips info message about unplugging and plugging back in.
- After saving the updated MtkChannelList.xml, I try updating the tv from the usb stick, I don't get any errors, but I don't see any updates. On the stick only the MtkChannelList.xml is changed after using chansort, all the others remain untouched

**Comment by alien2000a** [comment]
Dear Horst,

I'm sure you are fed up with Philips Channel 125 version. But I have a few theories to overcome the difficulty and avoid having to live with favourites list only in the future.(which I never used before)

Channel 125 system brings read only numbers to channels as you said. but there might be a catch to it. Before the Tv searches for satellites, it asks which search order  is preferred and provides certain options according to each country. For Turkey, for example it provides the option to sort it according to certain Settop Box, IPTV opertors and TKGS. TKGS is the system which automatically updates channel frequencies when changed/added WITHOUT changing the channel number. Same with other Cable Providers. Turkcell TV+, Tivibu etc have their fixed sort orders to match their own decoder boxes. they are trying to standardize the numbering I guess. So my theory is according to which provider you choose (including the TKGS) the numbers are fixed. That is probably why Chansort cannor change them or even if it changes the TV refuses the changes.

There might be a way around it and I'm not sure if this is the same case with other countries but it is worth a try. There is another option to search for channels with "no providers". This way the search behaves like a standard satellite search as it is being done in LG etc. So in theory, this channel list if exported should not have any set numbers. I can extract all different lists for you to compare and see if my theory works out.

One other thing I realized was that there are more than 1 file which has the channel list. 

1- Mtkchannellist.xml (Under \channellist)
2-atvchannellist.xml (Under \channellist)
3-DVBSall.xml (Under \channellist\s2channellib)
4- And there is the suspicious usr folder with no data in it. I assume this is somehow a user date folder for customized lists?

I will focus on the first three on the above list. maybe my findings may give you further clues about how to solve this mystery.

I edited item 1 only with the previous chansort version. Uploaded to TV, no changes were reflected on the TV
Then I defaulted item 1 back and edited item 2. Uploaded to TV, no changes were reflected on the TV
Similarly defaulted item 2 back and edited item 3. Uploaded to TV, no changes were reflected on the TV

This made me think whether all three or maybe any two in combination should be exactly the same to be able to reflect the changes to the TV.

So in short we could test,

- Different file extracts from the Tv as "different operator ordering" and one with "No ordering" to see if "No ordering" unblocks the channel number editing
- Try to reflect changes to all three files and in pairs if that fails to see if it works.

I'm more than happy to provide files from my TV and also test files back on TV if you develop new updates for chansort.

Many thanks.

**Comment by PredatH0r** [comment]
Hi @alien2000a, thanks for your detailed feedback!

I don't have any insider knowledge about the TV manufacturer industry, but it seems like Philips TVs with ChannelMap_120 and 125 are using a newer "platform" (=tuner hardware and firmware) from MediaTek, which now exports a MtkChanneList.xml. 
Sony seems to be using the same platform for the Google TV models and the contents of their "sdb.xml" are the same MediaTek format.

From what I can tell, the new TVs completely ignore the "atv\_channel\_list.xml" (analog channels) and "DVB\*.xml" files (digital) and only care about the newer MediaTek file and the Favorite.xml.

The MediaTek xml contains two channel lists. The one in the \<service\_list\_infos\> is readable XML and can be edited with ChanSort, but the TV seems to ignore that during the import. 
The second list is inside the \<internal\>\<scan\> block. It's a base64 encoded BLOB serialized by some Java code in the firmware, which can only be read by the exact same version of this Java code. The data inside that BLOB seems to be compressed by a proprietary MediaTek algorithm, with no publibly available method to decompress/decypher that data.

A user of a Sony TV sent me an original list and a list after swapping two channels through his TV's menu. Aside from the changes in the readable XML there was also a change in the \<scan\> BLOB, which seems to be the primary source for the TV when it imports the list. 

Unfortunately I have no way to modify that BLOB. Philips users can at least edit the Favorite.xml, which the TV imports as intended. Sony Bravia 7 (Google TV) users are completely out of luck, because the TV exports only the MediaTek file.

**Comment by andrei-lazarov** [comment]
Hello, thanks for the updates and progress on the 120 and 125 versions.

> Philips users can at least edit the Favorite.xml, which the TV imports as intended.

As far as I can tell editing the Favorites list only works on DVB-S right now.

I have the same TV with version 125 and DVB-C channels. I tried the 3 latest ChanSort versions to no luck.

- 10-15 gives me the same error as above: `MtkChannelList.xml doesn't contain a matching channel for @12: DIGI Sport 1` (first channel), regardless of the file I try to open and the program is blank.
<img width="455" alt="Screenshot2" src="https://github.com/user-attachments/assets/694cafa4-2326-485a-9d16-cfba8f06353b">

- the two older versions open the channel list from MtkChannelList.xml but not the Favorites list
<img width="455" alt="Screenshot 2024-11-13 211335" src="https://github.com/user-attachments/assets/fdddc19f-45a1-4d81-99ce-63cdec66c419">

.

Here is my list: [PhilipsDVBC.zip](https://github.com/user-attachments/files/17738436/PhilipsDVBC.zip)

Do you know what might cause this behavior?
I will take a look at the code. Let me know if you need more testing.
Thanks!

**Comment by Gabinger** [comment]
Leider die gleiche Fehlermeldung bei mir mit einem PHILIPS 48OLED809 4K OLED.
Gibt es inzwischen vielleicht eine Lösung??
